api/postgres/postgres.go

```diff
@@ -725,3 +746,65 @@ func ValueForFieldType(typ string, field string) string {
 		return fmt.Sprintf("\"%s\"", field)
 	}
 }
+// IsValidType validates the string to make sure it is a valid supported type
+func IsValidType(pType string) bool {
+	switch pType {
+		case dataTypeText:
+			return true    
+		case dataTypeDouble:  
+			return true 
+		case dataTypeFloat:   
+			return true 
+		case dataTypeVector:  
+			return true 
+		case dataTypeGeometry:
+			return true 
+		case dataTypeInteger: 
+			return true 
+		case dataTypeDate:   
+			return true  
+		default:
+			return false 
+	}
+}
+// IsColumnType can be use to check columns potential types
+func (d *Database)IsColumnType(tableName string, variable *model.Variable, colType string) (bool, error) {
```

Comment:
It may be better to put this in the mode/storage/postgres package somewhere and expose it via the storage interface